File: android/app/src/main/java/com/laneshadow/ui/atoms/ThemeBottomSheetInput.kt
package com.laneshadow.ui.atoms

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier

@Composable
fun ThemeBottomSheetInput(
    value: String,
    onValueChange: (String) -> Unit,
    modifier: Modifier = Modifier,
    label: String? = null,
    placeholder: String = "",
    enabled: Boolean = true,
    isError: Boolean = false,
    leftIconName: String? = null,
    rightIconName: String? = null,
) {
    ThemeInput(
        value = value,
        onValueChange = onValueChange,
        modifier = modifier,
        label = label,
        placeholder = placeholder,
        enabled = enabled,
        isError = isError,
        leftIconName = leftIconName,
        rightIconName = rightIconName,
    )
}

File: ios/LaneShadow/Views/Atoms/ThemeBottomSheetInput.swift
import SwiftUI

struct ThemeBottomSheetInput: View {
    let label: String?
    @Binding var text: String
    let placeholder: String
    let leftIconName: String?
    let rightIconName: String?
    let isError: Bool
    let isEditable: Bool

    init(
        label: String? = nil,
        text: Binding<String>,
        placeholder: String = "",
        leftIconName: String? = nil,
        rightIconName: String? = nil,
        isError: Bool = false,
        isEditable: Bool = true
    ) {
        self.label = label
        _text = text
        self.placeholder = placeholder
        self.leftIconName = leftIconName
        self.rightIconName = rightIconName
        self.isError = isError
        self.isEditable = isEditable
    }

    var body: some View {
        ThemeInput(
            label: label,
            text: $text,
            placeholder: placeholder,
            leftIconName: leftIconName,
            rightIconName: rightIconName,
            isError: isError,
            isEditable: isEditable
        )
        .submitLabel(.done)
    }
}
